python/modules/IcePy/PySliceLoader.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "PySliceLoader.h"
+#include "Types.h"
+
+#include <cassert>
+
+using namespace std;
+
+IcePy::PySliceLoader::PySliceLoader(PyObject* sliceLoader) : _sliceLoader(sliceLoader)
+{
+    assert(_sliceLoader);
+    assert(_sliceLoader != Py_None);
+
+    if (!PyCallable_Check(_sliceLoader))
+    {
+        throw Ice::InitializationException{__FILE__, __LINE__, "sliceLoader must be a callable"};
+    }
+}
+
+IcePy::PySliceLoader::~PySliceLoader()
+{
+    // Often called when the Python interpreter is finalizing.
+#if PY_VERSION_HEX >= 0x030D0000 // Python 3.13 and later
+    if (!Py_IsFinalizing())
+    {
+        Py_DECREF(_sliceLoader);
+    }
+#else
+    Py_DECREF(_sliceLoader);
```

Comment:
Not sure what to do here: we don't want to leak and we don't want to crash during Python shutdown.